wasmtime-wasi/src/host_impls.rs

```diff
@@ -121,7 +121,79 @@ macro_rules! rwlock_unlock {
     };
 }
 
-pub fn wasmtime_ssp_proc_exit(rval: wasm32::__wasi_exitcode_t) {
+pub fn wasmtime_ssp_args_get(
+    argv_environ: &mut host::argv_environ_values,
+    argv: &mut [*mut host::char],
+    argv_buf: &mut [host::char],
+) -> host::__wasi_errno_t {
+    for i in 0..(*argv_environ).argc {
```

Comment:
I might be wrong here, but since `argv_environ` is now a mutable reference, isn't explicit dereferencing it now unnecessary?